modules/openapi-generator/src/main/java/org/openapitools/codegen/config/CodegenConfigurator.java

```diff
@@ -643,6 +654,38 @@ public Context<?> toContext() {
         return new Context<>(specification, generatorSettings, workflowSettings);
     }
 
+    /*
+     * ODI-41 : Fix usage of x-protobuf-index for $ref fields
+     * This function automatically detect $ref that aren't part of an allOf and turn them into an allOf pattern
+     */
+    private void updateInputSpecReference() {
+        try {
+            String fileContent = new String(Files.readAllBytes(Paths.get(this.inputSpec)));
+            Pattern pattern = Pattern.compile("\\n\\s*(?<!-\\s?)\\$ref\\s?:.*\\n(?![\\s\\S]*components:)");
+            Matcher matcher = pattern.matcher(fileContent);
+
+            // Check if there's any $ref first
+            if (matcher.find()) {
+                File newInputSpec = new File("src/main/resources/codegen/temporaryInputSpec.yaml");
```

Comment:
Avoid to use/modify a real file by using a temp file :
https://mkyong.com/java/how-to-create-temporary-file-in-java/

(not forget to delete it after)

Reply:
Done.